poutyne/framework/model.py

```diff
@@ -573,9 +573,20 @@ def _fit_generator_n_batches_per_step(self, epoch_iterator, callback_list, batch
             train_step_iterator.epoch_metrics = self._get_epoch_metrics()
 
             if valid_step_iterator is not None:
+                valid_begin_time = timeit.default_timer()
+
+                callback_list.on_valid_begin({})
                 self._validate(valid_step_iterator)
+
+                valid_total_time = timeit.default_timer() - valid_begin_time
```

Comment:
The total time should include the time to obtain the epoch metrics (i.e. the call to `_get_epoch_metrics` below). Thank you for modifying at the 3 places it occurs in Model.

Reply:
4e82fe4